### PR TITLE
Fixes #25419: Add tests for the case where there is conflicts

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/25419

Just adding tests to see the outputs on conflicts. 

There is a lot of duplicated data in these outputs, it clutters it and should be removed. 
And there is no need to the json part when the user asked for html (I'm not even sure why we have that output). If the user wants json, he should not ask for html.

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/test/resources/api/api_groups.yml
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/test/resources/api/api_groups.yml:102]: this is the only difference, it could go on the first part and that part be removed

**Review by clarktsiory — APPROVED** [review]
LGTM !

**Comment by fanf** [comment]
Setting a can't merge for now, to avoid having to change tests all around plugins until the beta is released

File: webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TestRestFromFileDef.scala
Comment (clarktsiory) [reply webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TestRestFromFileDef.scala:115]: The by feature directory solution seems also a simpler solution, from our usage is seems more convenient than having to write imperative fixture setup code...
Also a more elaborate solution could be to parameterize all the mocks with the test data/fixtures. I am thinking about a structure which would make use of overrides like this :
```scala
class MockNode(nodeFixtures: NodeFixtures) {
  object nodeFactRepo {...}
}

// In Rudder
trait NodeFixtures {
  val nodeFacts: Map[NodeId, NodeFact] = ...
}
trait RestNodeFixtures extends RestNodeFixtures {
  // specific to rudder-rest
}

// In plugins
trait SystemUpdatesNodeFixtures extends NodeFixtures {
  // specific to rudder-rest
}
```
I think keeping a declarative approach for fixtures (with value overrides) would avoid the pain of understanding the imperative one

Comment (fanf) [reply webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TestRestFromFileDef.scala:115]: it could/seems to be the correct long term solution (not sure why traits, thought, that looks like just plain old data at different level that just need to be given at class instanciation ; we could even just have them in the mock class object and use them as default value for the param. It's simple to change data in scala)

**Comment by fanf** [comment]
Updated version with the simple case with a dedicated directory for the feature. 
So this approch shouldn't change anything on plugins and it can be merged once reviewed.

**Review by clarktsiory — APPROVED** [review]
LGTM ! The new by-directory test looks fine and they run successfully on my machine, but the CI seems to be capricious

**Pull request merged**
